core/consensus/babe/impl/babe_impl.cpp

```diff
@@ -56,27 +56,57 @@ namespace kagome::consensus {
     BOOST_ASSERT(hasher_);
     BOOST_ASSERT(log_);
 
-    NextEpochDescriptor epoch_0_and_1_digest;
-    epoch_0_and_1_digest.randomness = genesis_configuration_->randomness;
-    epoch_0_and_1_digest.authorities =
-        genesis_configuration_->genesis_authorities;
-    epoch_storage_->addEpochDescriptor(0, epoch_0_and_1_digest);
-    epoch_storage_->addEpochDescriptor(1, epoch_0_and_1_digest);
+    NextEpochDescriptor init_epoch_desc;
+    init_epoch_desc.randomness = genesis_configuration_->randomness;
+    init_epoch_desc.authorities = genesis_configuration_->genesis_authorities;
+    [[maybe_unused]] bool init_epoch_desc_ok =
+        !!epoch_storage_->addEpochDescriptor(0, init_epoch_desc)
+        && !!epoch_storage_->addEpochDescriptor(1, init_epoch_desc);
+
+    BOOST_ASSERT(init_epoch_desc_ok);
   }
 
-  void BabeImpl::runGenesisEpoch() {
-    Epoch genesis_epoch;
-    genesis_epoch.epoch_index = 0;
-    genesis_epoch.authorities = genesis_configuration_->genesis_authorities;
-    genesis_epoch.randomness = genesis_configuration_->randomness;
-    genesis_epoch.epoch_duration = genesis_configuration_->epoch_length;
-    genesis_epoch.start_slot = 0;
+  void BabeImpl::start() {
+    auto &&[best_block_number, best_block_hash] = block_tree_->deepestLeaf();
+    log_->debug("Babe run on block with number {} and hash {}",
+                best_block_number,
+                best_block_hash);
 
-    next_slot_finish_time_ =
-        clock_->now() + genesis_configuration_->slot_duration;
+    if (__builtin_expect(best_block_number == 0, false)) {
```

Comment:
is it some hint to compiler?

Reply:
Yep. But it will be removed any way, because it once-call method.